### PR TITLE
Fix modifying of frozen strings via config files and tests

### PR DESCRIPTION
**Pull request by KrauseFx**
Fixes https://github.com/fastlane/fastlane/issues/8548

### CODE REVIEW

**Comment by fastlane-bot-helper** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | File|
--- | ---|
FastlaneCore FastlaneCore::ConfigurationFile Properly loads and handles various configuration files properly loads boolean values | <a href='https://github.com/fastlane/fastlane/blob/53d369266260c74351fe388403d4d465d7dfbcaa/./fastlane_core/spec/configuration_file_spec.rb'>configuration_file_spec.rb</a>|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Comment by KrauseFx** [comment]
Oh wow, interesting, so turns out in 

**Ruby 2.3.0**
````
irb(main):003:0> false.dup
NoMethodError: undefined method `dupe' for false:FalseClass
```
**while in Ruby 2.4.0**
```
irb(main):003:0> false.dup
=> false
```

**Comment by KrauseFx** [comment]
Alright, updated to work with all versions of Ruby 👍

**Review by hjanuschka — APPROVED** [review]

**Comment by giginet** [comment]
By the way in current CI environments, the tests run only on Ruby 2.3.1.
Why don't we run specs on the multiple versions?
(We should run specs on Ruby2.3.x, 2.4.x and 2.0.0(Apple system default))

**Comment by hjanuschka** [comment]
we had that topic a few times i think right now circle does not support matrix runs, @DanToml  still right?

File: fastlane_core/spec/configuration_file_spec.rb [hunk 1]
```diff
@@ -57,6 +57,30 @@
...
+      it "supports modifying of frozen strings too" do
+        # Test that a value can be modified (this isn't the case by default if it's set via ENV)
+        app_identifier = "com.krausefx.yolo"
+        ENV["SOMETHING_RANDOM_APP_IDENTIFIER"] = app_identifier
+        config = FastlaneCore::Configuration.create(options, {})
+        config.load_configuration_file('ConfigFileEnv')
+        expect(config[:app_identifier]).to eq(app_identifier)
+        config[:app_identifier].gsub!("yolo", "yoLiveTwice")
+        expect(config[:app_identifier]).to eq("com.krausefx.yoLiveTwice")
+        ENV.delete("SOMETHING_RANDOM_APP_IDENTIFIER")
```
Comment (mfurtak) [inline fastlane_core/spec/configuration_file_spec.rb]: For unit tests we have a helper, `with_env_values` that will ensure that we don't forget to delete the ENV value at the end. Let's keep using that 👍 

```ruby
with_env_values('SOMETHING_RANDOM_IOS_VERSION' -> ios_version) do
  config = FastlaneCore::Configuration.create(options, {})
  config.load_configuration_file('ConfigFileEnv')
  expect(config[:ios_version]).to eq(ios_version)
  config[:ios_version].gsub!(".1", ".2")
  expect(config[:ios_version]).to eq("9.2")
end
```

File: fastlane_core/spec/configuration_file_spec.rb [hunk 2]
```diff
@@ -57,6 +57,30 @@
...
+      it "supports modifying of frozen strings that are returned via blocks too" do
+        # Test that a value can be modified (this isn't the case by default if it's set via ENV)
+        ios_version = "9.1"
+        ENV["SOMETHING_RANDOM_IOS_VERSION"] = ios_version
+        config = FastlaneCore::Configuration.create(options, {})
+        config.load_configuration_file('ConfigFileEnv')
+        expect(config[:ios_version]).to eq(ios_version)
+        config[:ios_version].gsub!(".1", ".2")
+        expect(config[:ios_version]).to eq("9.2")
+        ENV.delete("SOMETHING_RANDOM_IOS_VERSION")
```
Comment (mfurtak) [inline fastlane_core/spec/configuration_file_spec.rb]: Let's use `with_env_values(hash, &block)`

**Comment by DanToml** [comment]
@hjanuschka still correct

**Pull request merged**